junit5-extension/src/test/java/io/kroxylicious/testing/kafka/junit5ext/TemplateTest.java

```diff
@@ -160,10 +167,13 @@ void testTuples(@ConstraintsMethodSource(value = "tuples", clazz = TemplateTest.
 
         @AfterAll
         void afterAll() {
-            assertThat(observedTuples).isEqualTo(Set.of(
+            var expected = new HashSet<>(Set.of(
                     List.of(1, 1),
-                    List.of(3, 1),
-                    List.of(3, -1)));
+                    List.of(3, 1)));
+            if (ZOOKEEPER_AVAILABLE) {
+                expected.add(List.of(3, -1));
+            }
+            assertThat(observedTuples).isEqualTo(expected);
```

Comment:
does `isEqualTo` care about the type of the set? Just wondering if this would be better expressed as 
```suggestion
            assertThat(observedTuples).containsExactlyInAnyOrderElementsOf(expected);
```

I wonder if that might be a bit more forgiving as I think this is one of the more temperamental tests